src/schema/opportunity.ts

```diff
@@ -1920,4 +2133,272 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
+      ) {
+        const loc = parent.candidatePreference.location[0];
+        return [loc.city, loc.subdivision, loc.country]
+          .filter(Boolean)
+          .join(', ');
+      }
+
+      // Fallback to user geo flags from parent
+      if (parent.user?.flags) {
+        return [
+          parent.user.flags.city,
+          parent.user.flags.subdivision,
+          parent.user.flags.country,
+        ]
+          .filter(Boolean)
+          .join(', ');
+      }
+
+      return null;
+    },
+    company: async (parent: {
+      userId: string;
+      latestExperience?: UserExperience;
+    }): Promise<{ name: string; favicon?: string } | null> => {
+      if (!parent.latestExperience) {
+        return null;
+      }
+
+      // Use COALESCE logic: prefer company name over customCompanyName
+      let companyName: string | null = null;
+
+      // Try to get company name from the loaded relation
+      const companyRelation = await parent.latestExperience.company;
+      if (companyRelation?.name) {
+        companyName = companyRelation.name;
+      } else if (parent.latestExperience.customCompanyName) {
+        companyName = parent.latestExperience.customCompanyName;
+      }
+
+      if (!companyName) {
+        return null;
+      }
+
+      // TODO: Fetch favicon from company enrichment service
+      return {
+        name: companyName,
+        favicon: undefined,
+      };
+    },
+    lastActivity: async (
+      parent: { userId: string },
+      _,
+      ctx: Context,
+    ): Promise<Date | null> => {
+      if (!parent.userId) {
+        return null;
+      }
+      return await ctx.dataLoader.userLastActive.load({
+        userId: parent.userId,
+      });
+    },
+    topTags: async (
+      parent: { userId: string },
+      _,
+      ctx: Context,
+    ): Promise<string[] | null> => {
+      const tags = await ctx.con
+        .createQueryBuilder()
+        .select('pk.keyword', 'keyword')
+        .addSelect('COUNT(*)', 'count')
+        .from('user_post', 'up')
+        .innerJoin('post_keyword', 'pk', 'pk.postId = up.postId')
+        .where('up.userId = :userId', { userId: parent.userId })
+        .andWhere('up.hidden = false')
+        .groupBy('pk.keyword')
+        .orderBy('count', 'DESC')
+        .limit(5)
+        .getRawMany<{ keyword: string }>();
+
+      return tags.length > 0 ? tags.map((t) => t.keyword) : null;
+    },
+    recentlyRead: async (
+      parent: { userId: string },
+      _,
+      ctx: Context,
+    ): Promise<string[] | null> => {
+      const posts = await ctx.con
+        .createQueryBuilder()
+        .select('up.postId', 'postId')
+        .from('user_post', 'up')
+        .where('up.userId = :userId', { userId: parent.userId })
+        .andWhere('up.hidden = false')
+        .orderBy('up.createdAt', 'DESC')
+        .limit(3)
+        .getRawMany<{ postId: string }>();
+
+      return posts.length > 0 ? posts.map((p) => p.postId) : null;
+    },
+    activeSquads: async (
```

Comment:
honestly issue with this is that it makes multiple queries instead of one gql one as usual, maybe it can move it to graphorm layer?